src/CommunityToolkit.Maui/Extensions/PopupExtensions.shared.cs

```diff
@@ -0,0 +1,122 @@
+	/// <param name="token"><see cref="CancellationToken"/></param>
+	/// <returns>Popup Result</returns>
+	public static Task<IPopupResult<TResult>> ShowPopupAsync<TResult>(this Page page, View view, IPopupOptions? options = null, CancellationToken token = default)
+	{
+		ArgumentNullException.ThrowIfNull(page);
+
+		return page.Navigation.ShowPopupAsync<TResult>(view, options, token);
+	}
+
+	/// <summary>
+	/// Opens a popup with the specified options.
+	/// </summary>
+	/// <param name="navigation">Popup parent</param>
+	/// <param name="view">Popup content</param>
+	/// <param name="options"><see cref="IPopupOptions"/></param>
+	/// <param name="token"><see cref="CancellationToken"/></param>
+	/// <returns>Popup Result</returns>
+	public static async Task<IPopupResult<TResult>> ShowPopupAsync<TResult>(this INavigation navigation, View view, IPopupOptions? options = null, CancellationToken token = default)
+	{
+		var result = await ShowPopupAsync(navigation, view, options, token);
+
+		return result switch
+		{
+			PopupResult<TResult> popupResult => popupResult,
+			IPopupResult => new PopupResult<TResult>(null, result.WasDismissedByTappingOutsideOfPopup),
+			_ => throw new NotSupportedException($"PopupResult type {typeof(TResult)} is not supported")
+		};
+	}
+
+	/// <summary>
+	/// Shows a popup with the specified options.
+	/// </summary>
+	/// <param name="page">Current page</param>
+	/// <param name="view">Popup content</param>
+	/// <param name="options"><see cref="IPopupOptions"/></param>
+	/// <param name="token"><see cref="CancellationToken"/></param>
+	/// <returns><see cref="PopupResult"/></returns>
+	public static Task<IPopupResult> ShowPopupAsync(this Page page, View view, IPopupOptions? options = null, CancellationToken token = default)
+	{
+		ArgumentNullException.ThrowIfNull(page);
+
+		return ShowPopupAsync(page.Navigation, view, options, token);
+	}
+
+	/// <summary>
+	/// Shows a popup with the specified options.
+	/// </summary>
+	/// <param name="navigation">Popup parent</param>
+	/// <param name="view">Popup content</param>
+	/// <param name="options"><see cref="IPopupOptions"/></param>
+	/// <param name="token"><see cref="CancellationToken"/></param>
+	/// <returns><see cref="PopupResult"/></returns>
+	public static async Task<IPopupResult> ShowPopupAsync(this INavigation navigation, View view, IPopupOptions? options, CancellationToken token = default)
+	{
+		ArgumentNullException.ThrowIfNull(navigation);
+		ArgumentNullException.ThrowIfNull(view);
+
+		token.ThrowIfCancellationRequested();
+
+		TaskCompletionSource<IPopupResult> taskCompletionSource = new();
+
+		var popupPage = new PopupPage(view, options ?? PopupOptions.Empty);
+		popupPage.PopupClosed += HandlePopupClosed;
+
+		await navigation.PushModalAsync(popupPage, false).WaitAsync(token);
+		return await taskCompletionSource.Task.WaitAsync(token);
+
+		void HandlePopupClosed(object? sender, IPopupResult e)
+		{
+			taskCompletionSource.SetResult(e);
```

Comment:
The handler isn't static, so just to be safe let's unsubscribe it
```suggestion
			taskCompletionSource.SetResult(e);
			popupPage.PopupClosed -= HandlePopupClosed;
```

Reply:
Sounds good 👍